File: android_samples/33 ToDoListMVP/app/src/main/java/com/mnegi/todolistmvp/views/RecyclerViewAdapter.java
package com.mnegi.todolistmvp.views;

import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import com.mnegi.todolistmvp.R;
import com.mnegi.todolistmvp.pojo.TaskItem;
import java.util.List;

public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder>
{
    List<TaskItem> list;
    CallBackInterface callBackInterface;

    public RecyclerViewAdapter(List<TaskItem> list, CallBackInterface callBackInterface) {
        this.list = list;
        this.callBackInterface = callBackInterface;
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.layout_rv_item , parent,false);
        View_Holder holder = new View_Holder(view);
        return holder;
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
        ((View_Holder) holder).title.setText(list.get(position).getTaskName());
        ((View_Holder) holder).description.setText(list.get(position).getTaskDescription());
        ((View_Holder) holder).editBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                callBackInterface.editItem(list.get(position).getID());
            }
        });
        ((View_Holder) holder).deleteBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                callBackInterface.deleteItem(list.get(position).getID());
            }
        });
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    @Override
    public int getItemViewType(int position) {
        return super.getItemViewType(position);
    }

    public class View_Holder extends RecyclerView.ViewHolder
    {
        TextView title;
        TextView description;
        ImageView editBtn, deleteBtn;

        public View_Holder(View view) {
            super(view);
            title = (TextView) view.findViewById(R.id.titletxt);
            description = (TextView) view.findViewById(R.id.descriptiontxt);
            editBtn = (ImageView) view.findViewById(R.id.editBtn);
            deleteBtn = (ImageView) view.findViewById(R.id.deleteBtn);
        }
    }
}

File: android_samples_kotlin/06 RoomDBExample/app/src/main/java/com/msnegi/roomdbexample/RecyclerViewAdapter.kt
package com.msnegi.roomdbexample

import androidx.recyclerview.widget.RecyclerView
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import com.msnegi.roomdbexample.roomdb.Item

class RecyclerViewAdapter(var list: List<Item?>?, var callBackInterface: CallBackInterface) :
    RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): RecyclerView.ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.layout_rv_item, parent, false)
        return View_Holder(view)
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        (holder as View_Holder).title.text = list!![position]!!.item_id
        holder.description.text = list!![position]!!.descripton
        holder.editBtn.setOnClickListener {
            callBackInterface.editItem(
                list!![position]
            )
        }
        holder.deleteBtn.setOnClickListener { callBackInterface.deleteItem(list!![position]) }
    }

    override fun getItemCount(): Int {
        return list!!.size
    }

    override fun getItemViewType(position: Int): Int {
        return super.getItemViewType(position)
    }

    inner class View_Holder(view: View) : RecyclerView.ViewHolder(view) {
        var title: TextView
        var description: TextView
        var editBtn: ImageView
        var deleteBtn: ImageView

        init {
            title = view.findViewById<View>(R.id.titletxt) as TextView
            description = view.findViewById<View>(R.id.descriptiontxt) as TextView
            editBtn = view.findViewById<View>(R.id.editBtn) as ImageView
            deleteBtn = view.findViewById<View>(R.id.deleteBtn) as ImageView
        }
    }
}